bolt-design-system/bolt
# Add Google Analytics Autolink Helper

## Jira
http://vjira2:8080/browse/WWWD-2988

## Summary
Adds a new Google analytics `autolink` helper package to make it easy to wire up Bolt components with Google Analytics Linker plugin so components with external domain URLs (regardless on how a component renders, what browser a component is rendering in, and whether or not a component renders / re-renders) are much easier to track.

## Details
- While no changes to any existing Bolt components was technically required to get this wired up (since this works by hooking into the `rendered` custom event every Bolt component emits when initially rendering + re-rendering when data changes), this PR:
- Formalizes this integration point (ensure there isn't any unexpected API changes)
- Makes sure this approach continues to work as expected
- Adds testing coverage to verify a variety of different use cases / situations / configuration options
- Provides overall docs on usage + configuration 

## How to test
- Confirm the build + new tests added all pass on Travis
- Read through the docs to make sure usage and configuration all make sense

Three things in particular @danielamorse I'd like to get your opinion on... 

1. Where exactly in the monorepo codebase do you think this package makes the most sense to live in? I settled on putting this in a new `packages/analytics/autolink` folder with the idea that additional analytics-specific packages will be added, however I also considered having this live in the components folder (despite this package only applying component behavior vs being a "traditional" component).

2. Given that this isn't a "traditional" component, this also unfortunately means it doesn't have docs that automatically show up in the docs / Pattern Lab which I find to be problematic.

3. Thoughts on this being packaged up as a super simple custom element? ex. `<bolt-autolink>`? That could solve the need to have a proper home for this in addition providing a declarative way to configure the domains being tracked (ex. allow domains to declaratively get passed along via the custom element vs only providing a global JS config option which imho, really doesn't feel like it scales...)
